### PR TITLE
Rewards summary should now be cleared when wallet is restored

### PR DESCRIPTION
**Pull request by jasonrsadler**
Core implementation: https://github.com/brave/brave-core/pull/777

### CODE REVIEW

File: src/bat_publishers.cc [hunk 1]
```diff
@@ -745,6 +753,7 @@ void BatPublishers::setBalanceReport(ledger::PUBLISHER_MONTH month,
 
   report_balance.total_ = total;
   state_->monthly_balances_[GetBalanceReportName(month, year)] = report_balance;
+
```
Comment (NejcZdovc) [inline src/bat_publishers.cc]: remove this line

File: src/ledger_impl.cc [hunk 2]
```diff
@@ -593,7 +593,7 @@ void LedgerImpl::RecoverWallet(const std::string& passPhrase) const {
 
 void LedgerImpl::OnRecoverWallet(ledger::Result result, double balance, const std::vector<braveledger_bat_helper::GRANT>& grants) {
   std::vector<ledger::Grant> ledgerGrants;
-
+  LOG(ERROR) << "RECOVERING===========";
```
Comment (NejcZdovc) [inline src/ledger_impl.cc]: remove this log

File: src/ledger_impl.cc [hunk 3]
```diff
@@ -604,6 +604,8 @@ void LedgerImpl::OnRecoverWallet(ledger::Result result, double balance, const st
     ledgerGrants.push_back(tempGrant);
   }
 
+  bat_publishers_->clearAllBalanceReports();
```
Comment (NejcZdovc) [inline src/ledger_impl.cc]: clear reports only if recover is successful

File: src/bat_publishers.cc (hunk 1 above)
Comment (jasonrsadler) [reply src/bat_publishers.cc]: Done

File: src/ledger_impl.cc (hunk 2 above)
Comment (jasonrsadler) [reply src/ledger_impl.cc]: Done

File: src/ledger_impl.cc (hunk 3 above)
Comment (jasonrsadler) [reply src/ledger_impl.cc]: Done

**Pull request merged**